src/main/java/org/example/learningsystem/core/featureflags/service/FeatureFlagsServiceImpl.java

```diff
@@ -0,0 +1,74 @@
+package org.example.learningsystem.core.featureflags.service;
+
+import org.example.learningsystem.core.featureflags.exception.FeatureFlagTypeMismatchException;
+import org.example.learningsystem.core.featureflags.config.FeatureFlagsProperties;
+import org.example.learningsystem.core.featureflags.dto.FlagDto;
+import org.springframework.context.annotation.Profile;
+import org.springframework.http.HttpHeaders;
+import org.springframework.retry.annotation.Recover;
+import org.springframework.retry.annotation.Retryable;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.HttpStatusCodeException;
+import org.springframework.web.client.RestClient;
+
+import static java.util.Objects.isNull;
+import static org.springframework.http.HttpStatus.NOT_FOUND;
+import static org.springframework.http.HttpStatus.OK;
+
+@Service
+@Profile("cloud")
+public class FeatureFlagsServiceImpl implements FeatureFlagsService {
+
+    private final FeatureFlagsProperties properties;
+    private final RestClient restClient;
+    private static final String BOOLEAN_FLAG_TYPE = "BOOLEAN";
+    private static final String EVALUATE_FLAG_URI = "%s/api/v2/evaluate/%s";
```

Comment:
order of static and non-static variables should be different
static variables are going first